cleanlab_studio/studio/studio.py

```diff
@@ -266,7 +268,30 @@ def poll_cleanset_status(self, cleanset_id: str, timeout: Optional[int] = None)
         Returns:
             After cleanset is done being generated, returns `True` if cleanset is ready to use, `False` otherwise.
         """
-        return clean_helpers.poll_cleanset_status(self._api_key, cleanset_id, timeout)
+        warnings.warn(
+            "Poll cleanset status method has been deprecated -- please use wait_for_cleanset_ready method instead.",
+            DeprecationWarning,
+        )
+
+        try:
+            clean_helpers.poll_cleanset_status(self._api_key, cleanset_id, timeout)
+            return True
+
+        except (TimeoutError, CleansetRunError):
+            return False
+
+    def wait_until_cleanset_ready(self, cleanset_id: str, timeout: Optional[float] = None) -> None:
+        """Blocks until cleanset is ready, timeout is reached, or cleanset errors.
```

Comment:
```suggestion
        """Blocks until a cleanset is ready or the timeout is reached.
```

Reply:
I think the error case should be highlighted in the first line of the description. If our server code is bug-free, this should never happen (it's never a user error, always an internal server error).